src/azure-cli/azure/cli/command_modules/monitor/_client_factory.py

```diff
@@ -5,69 +5,69 @@
 
 
 # MANAGEMENT CLIENT FACTORIES
-def cf_monitor(cli_ctx, _):
-    from azure.mgmt.monitor import MonitorManagementClient
+def cf_monitor(cli_ctx, **kwargs):
+    from azure.cli.core.profiles import ResourceType
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    return get_mgmt_service_client(cli_ctx, MonitorManagementClient)
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_MONITOR, **kwargs)
 
 
 def cf_alert_rules(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).alert_rules
+    return cf_monitor(cli_ctx).alert_rules
 
 
 def cf_alert_rule_incidents(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).alert_rule_incidents
+    return cf_monitor(cli_ctx).alert_rule_incidents
 
 
 def cf_autoscale(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).autoscale_settings
+    return cf_monitor(cli_ctx).autoscale_settings
 
 
 def cf_diagnostics(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).diagnostic_settings
+    return cf_monitor(cli_ctx).diagnostic_settings
 
 
 def cf_diagnostics_category(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).diagnostic_settings_category
+    return cf_monitor(cli_ctx).diagnostic_settings_category
 
 
 def cf_log_profiles(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).log_profiles
+    return cf_monitor(cli_ctx).log_profiles
 
 
 def cf_action_groups(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).action_groups
+    return cf_monitor(cli_ctx).action_groups
 
 
 def cf_activity_log_alerts(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).activity_log_alerts
+    return cf_monitor(cli_ctx).activity_log_alerts
 
 
 def cf_metrics(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).metrics
+    return cf_monitor(cli_ctx).metrics
 
 
 def cf_metric_def(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).metric_definitions
+    return cf_monitor(cli_ctx).metric_definitions
 
 
 def cf_activity_log(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).activity_logs
+    return cf_monitor(cli_ctx).activity_logs
 
 
 def cf_event_categories(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).event_categories
+    return cf_monitor(cli_ctx).event_categories
 
 
 def cf_metric_alerts(cli_ctx, _):
-    return cf_monitor(cli_ctx, _).metric_alerts
+    return cf_monitor(cli_ctx).metric_alerts
 
 
-def _log_analytics_client_factory(cli_ctx, _):
+def _log_analytics_client_factory(cli_ctx, **kwargs):
     from azure.mgmt.loganalytics import LogAnalyticsManagementClient
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    return get_mgmt_service_client(cli_ctx, LogAnalyticsManagementClient)
+    return get_mgmt_service_client(cli_ctx, LogAnalyticsManagementClient, **kwargs)
```

Comment:
>kwargs [](start = 76, length = 6)

Why do we need add "**kwargs" here?

Reply:
left the room for future in case we need to input sub or aux_sub in the future.